server/cmd/mmctl/commands/auth.go

```diff
@@ -164,7 +167,20 @@ func loginCmdF(cmd *cobra.Command, args []string) error {
 	allowInsecureSHA1 := viper.GetBool("insecure-sha1-intermediate")
 	allowInsecureTLS := viper.GetBool("insecure-tls-version")
 
-	url := strings.TrimRight(args[0], "/")
+	instanceURL := strings.TrimRight(args[0], "/")
+	_, urlErr := url.ParseRequestURI(instanceURL)
+	if urlErr != nil {
+		return fmt.Errorf("could not parse the instance url: %w", urlErr)
+	}
+
+	res, err := http.Get(instanceURL)
+	if err != nil {
+		return fmt.Errorf("could not get instance status: %w", err)
+	}
+	if res.StatusCode != 200 {
+		return fmt.Errorf("instance status code is not 200: %d", res.StatusCode)
+	}
```

Comment:
Instead of using plain HTTP, you should call `model.NewAPIv4Client` to construct a client and then call `GetPingWithOptions` to execute the ping.

Reply:
Okay